File: ClockAngle.c
#include <stdio.h>
#include <math.h>

void main() {
    int hour, minute;
    double hourAngle, minuteAngle, angle;

    printf("Enter hour: ");
    scanf("%d", &hour);

    printf("Enter minutes: ");
    scanf("%d", &minute);

    minuteAngle = minute * 6;               // 1 minute = 6 degrees
    hourAngle = (hour % 12) * 30 + minute * 0.5; // 1 hour = 30 degrees, 1 min = 0.5 degree

    angle = fabs(hourAngle - minuteAngle);

    if (angle > 180)
        angle = 360 - angle;

    printf("Angle between clock hands = %.2f degrees", angle);
}

File: ClockAngle.java
import java.util.Scanner;

public class ClockAngle {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter hour: ");
        int hour = sc.nextInt();

        System.out.print("Enter minutes: ");
        int minute = sc.nextInt();

        double minuteAngle = minute * 6;                    // 1 minute = 6 degrees
        double hourAngle = (hour % 12) * 30 + minute * 0.5; // 1 hour = 30 degrees, 1 minute = 0.5 degree

        double angle = Math.abs(hourAngle - minuteAngle);

        if (angle > 180)
            angle = 360 - angle;

        System.out.println("Angle between clock hands = " + angle + " degrees");

        sc.close();
    }
}